### ISSUE TITLE
Unable to use ! with the empty field value

### ISSUE THREAD

**Issue by codebased** [angular/angular.js#11756]
With the previous version I could use "!" for filtering records where field has null or empty value.

example: 

```javascript

ng-repeat="category in vm.model.categories | filter: {parentId: '!'}"

```

After changing to the latest, it is breaking.

Any suggestion?

**Comment by petebacondarwin** [comment]
This works for me with the latest on master: http://plnkr.co/edit/oURMfuBqYGBcMrXEyYsR?p=preview
and also with the latest release candidate: http://plnkr.co/edit/IAy2Eq6xTPRDpik6GSfw?p=preview

**Comment by gkalpak** [comment]
@codebased is probably referring to the 1.3.x branch.

There was a regression since 1.3.6, which has been fixed on the 1.4.x branch, but not on 1.3.x yet.
We have discussed (somewhere) that we should backport the fix (since it was a regression) and I have submitted PR #11617 against the 1.3.x branch (but it's not merged yet); hence it's still reproducible on 1.3.15.

@petebacondarwin, should I merge #11617 ?
I submitted the PR to make sure Travis is happy, but I am not sure if I should get a LGTM or not (since this is a backported fix).

**Comment by codebased** [comment]
@gkalpak @petebacondarwin  Yes it is   "version": "1.3.15",

**Comment by petebacondarwin** [comment]
LGTM

**Comment by gkalpak** [comment]
Hm... I just realized, @Narretz has already merged my PR since yesterday (9dd0fe3), so this should be fixed for 1.3.x as soon as v1.3.16 is out.

**Comment by Narretz** [comment]
Yes, I merged it because it was a regression. Github should probably send out notifications when subscribed issues are closed ...
